erts/emulator/beam/erl_bif_binary.c

```diff
@@ -782,55 +854,31 @@ static Sint bm_find_first_match(BMFindFirstState *state, BMData *bmd,
 	    ;
 	if (i < 0) { /* found */
 	    *reductions = reds;
-	    return j;
+	    *mpos = (Uint) j;
+	    *mlen = (Uint) blen;
```

Comment:
So, I played around with this, but couldn't find a simple way to get this done without changing the data types for BMData and BMFindFirstState to Uint as well.  The way that some of the loops have been previously written for the BM functions are designed for a signed integer that can terminate the loop once the number is negative.

Would you like me to refactor the BM related functions so that everything can be a Uint similar to how the AC functions work?